bionlp/preprocess/parser.py

```diff
@@ -0,0 +1,96 @@
+from abc import ABC, abstractmethod
+import json
+import os
+
+from bionlp.tags import IOB2_Tag
+
+class AbstractDatasetParser(ABC):
+    @abstractmethod
+    def parse_datasets(self, input_dir, output_dir):
+        raise NotImplementedError
+
+    @abstractmethod
+    def parse_file(self, input_file, output_file):
+        raise NotImplementedError
+
+
+class BratParser(AbstractDatasetParser):
+    # this is implemented in the BratParser class instead of the abstract parent class because,
+    # for Brat, we only want to regard the input files with file extension .ann
+    def parse_datasets(self, input_dir, output_dir):
+        # this methodology only expects files in the top level of the given input directory, not subdirs
+        _, _, input_files = next(os.walk(input_dir))
+        for input_file in input_files:
```

Comment:
`os.listdir(input_dir)` would return not only the files, but also the directories located in `input_dir`. I explicitly wanted only the files, so I chose this approach.